src/main/kotlin/mate/academy/Main.kt

```diff
@@ -1,7 +1,14 @@
 package mate.academy
 
+const val GREETING = "Please enter a name (or press Enter to quit): "
+
 fun greetUsers() {
-    // implement function
+    var userName: String
+    while (true) {
+        println(GREETING)
+        userName = readlnOrNull().toString()
+        if (userName == "") break else println("Hello, $userName!")
```

Comment:
This code works. However, consider to simplify it using kotlin functionality.
Please check these links
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/if-empty.html
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/is-null-or-empty.html

Note, you may use 'return' instead of 'break'